nixos/modules/services/security/crowdsec.nix

```
@@ -0,0 +1,867 @@
+                    ''
+                  ]
+                  ++ lib.optionals (tokenFile != null) [
+                    ''
+                      if [ ! -e "${tokenFile}" ]; then
+                        ${lib.getExe cscli} console enroll "$(cat ${tokenFile})" --name ${cfg.name}
+                      fi
+                    ''
+                  ];
+
+                script = pkgs.writeShellScriptBin "crowdsec-setup" (lib.strings.concatStringsSep "\n" scriptArray);
+              in
+              [
+                " " # This is needed to clear the ExecStartPre definitions from upstream
+                "${lib.getExe script}"
+                "${lib.getExe' cfg.package "crowdsec"} -c ${configFile} -t -error"
+              ];
+          };
+        };
+      };
+
+      systemd.tmpfiles.rules = (
+        [
+          "d ${stateDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${hubDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${confDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localScenariosDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localPostOverflowsDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localPostOverflowsS01WhitelistDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${parsersDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localParsersS00RawDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localParsersS01ParseDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localParsersS02EnrichDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${localContextsDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${notificationsDir} 0750 ${cfg.user} ${cfg.group} - -"
+          "d ${pluginDir} 0750 ${cfg.user} ${cfg.group} - -"
+        ]
+        ++ (map (
+          file: "L+ ${localScenariosDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localScenariosMap)
+        ++ (map (
+          file: "L+ ${localParsersS00RawDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localParsersS00RawMap)
+        ++ (map (
+          file: "L+ ${localParsersS01ParseDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localParsersS01ParseMap)
+        ++ (map (
+          file: "L+ ${localParsersS02EnrichDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localParsersS02EnrichMap)
+        ++ (map (
+          file: "L+ ${localPostOverflowsS01WhitelistDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localPostOverflowsS01WhitelistMap)
+        ++ (map (
+          file: "L+ ${localContextsDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localContextsMap)
+        ++ (map (
+          file: "L+ ${notificationsDir}/${builtins.baseNameOf file} - - - - ${file}"
+        ) localNotificationsMap)
+      );
+
+      users.users.${cfg.user} = {
+        name = lib.mkDefault cfg.user;
+        description = lib.mkDefault "CrowdSec service user";
+        isSystemUser = lib.mkDefault true;
+        group = lib.mkDefault cfg.group;
```

```suggestion
        group = cfg.group;
```
People should be using the options and get an error if not